src/components/DeleteModal.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC } from 'react';
+
+import Button from './Button';
+import { DeleteModalProps } from './types';
+
+import { useDeleteEmployeeMutation } from 'services/api';
+
+const DeleteModal: FC<DeleteModalProps> = (props) => {
+    const { setDelete, clickedId } = props;
+
+    const [deleteEmployee] = useDeleteEmployeeMutation();
+
+    const handleDeleteEmployee = (clickedEmployeeId: number) => {
+        deleteEmployee(clickedEmployeeId);
+        setDelete(false);
+    };
```

Comment:
define the function in the parent container and pass it as prop. make the state update too in the container itself. So props here becomes "handleDelete function and the clickedId"